### PR TITLE
fix: Form widget should include the following widgets in data property when added in form

### LINKED ISSUE #11188: [Bug]: Form widget should include the following widgets in data property when added in form
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The form widget does not include the multi-select, switch group widget selected values in form.data property. It only includes children widgets that expose the selected value in the `value`  key into the data property.
Input, Currency, Phone, select expose the entered/chosen value in `value`  property. But multi-select, switch group, etc are not exposing the `value`  property.

### Steps To Reproduce

1. Add a form widget in canvas
2. Add a multiselect widget in the form
3. choose a value on multiselect
4. form.data property doesn't have this value

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by wmdev0808**
The form widget does not include the multi-select, switch group widget selected values in form.data property. It only includes children widgets that expose the selected value in the value key into the data property.
Input, Currency, Phone, select expose the entered/chosen value in value property. But multi-select, switch group, etc are not exposing the value property.

> Widgets in the current scope:
- CheckboxGroupWidget
- MultiSelectTreeWidget
- MultiSelectWidgetV2
- SingleSelectTreeWidget
- SwitchGroupWidget

Fixes #11188 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Test A
- Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/11188-form-data-issue 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 83.61 **(-1.64)** | 61.76 **(-2.95)** | 73.33 **(0)** | 88.24 **(-2.35)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

**Comment by wmdev0808** [comment]
@sbalaji1192 ,
In Phone Input and Currency Input widget, 
Is excluding empty value by setting `value` meta prop to `undefined` valid or not?

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+describe("Form data", function() {
+  before(() => {
+    cy.addDsl(dsl);
+  });
+
+  it("CheckboxGroupWidget, MultiSelectTreeWidget, MultiSelectWidgetV2, SingleSelectTreeWidget, SwitchGroupWidget and PhoneInputWidget should have value props of which values are not null or undefined to be included into form data", function() {
+    // Check form data
+    cy.get("[data-testid='container-wrapper-vannrar7rd'] span", {
+      timeout: 25000,
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js:11]: why do we need to have such a long wait time?

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    })
+      .should("exist")
+      .and(($formData) => {
+        expect($formData).to.contain("CheckboxGroup");
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js:15]: we need to check select, Input and Currency widget as well.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js (hunk 1 above)
Comment (wmdev0808) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js:11]: Showing off text widget bound to `form.data` takes such a long time

Comment (sbalaji1192) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js:11]: I don't think it should take that much time. If it takes that much time on cypress , it should take the same amount of time for the User as well. Did you check why there is a delay?

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        expect($formData).to.contain("MultiTreeSelect");
+        expect($formData).to.contain("MultiSelect");
+        expect($formData).to.contain("TreeSelect");
+        expect($formData).to.contain("SwitchGroup");
+        expect($formData).to.contain("PhoneInput");
+        expect($formData).to.contain("Input");
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/FormData_spec.js:21]: select is still missing

**Review by sbalaji1192 — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@wmdev0808 Tested this PR and please find the status below

1.CheckboxGroupWidget - **Verified** 
2.MultiSelectTreeWidget - **Verified**
3.MultiSelectWidgetV2 - **Not Verified** 
4.SingleSelectTreeWidget - **Verified** 
5.SwitchGroupWidget - **Verified** 

Please find the issues observed:

1: The default Currency data are not displayed in the Form.data - It is observed that the form.data displays only when the currency widget has some input 

https://loom.com/share/8ddc883833a047cc8906f69dcfa7b5a6

2. Currently able to verify the MultiSelect property data in the form.data but not the **MultiSelectWidgetV2** as there is **no MultiSelectV2 in the DP**

**Comment by wmdev0808** [comment]
> @wmdev0808 Tested this PR and please find the status below
> 
> 1.CheckboxGroupWidget - **Verified** 2.MultiSelectTreeWidget - **Verified** 3.MultiSelectWidgetV2 - **Not Verified** 4.SingleSelectTreeWidget - **Verified** 5.SwitchGroupWidget - **Verified**
> 
> Please find the issues observed:
> 
> 1: The default Currency data are not displayed in the Form.data - It is observed that the form.data displays only when the currency widget has some input
> 
> https://loom.com/share/8ddc883833a047cc8906f69dcfa7b5a6
> 
> 2. Currently able to verify the MultiSelect property data in the form.data but not the **MultiSelectWidgetV2** as there is **no MultiSelectV2 in the DP**

Hi @YogeshJayaseelan 
1: It's expected behavior with CurrencyInput
2: MultiSelect in entity explorer is just MultiSelectWidgetV2

**Comment by YogeshJayaseelan** [comment]
@wmdev0808 Tested this PR and it is working as expected 

1.CheckboxGroupWidget - Verified
2.MultiSelectTreeWidget - Verified
3.MultiSelectWidgetV2 - Verified (Mutliselect is itself an MutliSelectV2)
4.SingleSelectTreeWidget - Verified
5.SwitchGroupWidget - Verified

Note: The form.data will not display the currency input type when there is no default value

**Pull request merged**